src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -146,6 +153,12 @@ private void CloseInternal()
                 // Suppress finalizer if the read buffer was returned.
                 GC.SuppressFinalize(this);
             }
+
+            // Set the status to disposed. If it was opened before, log ConnectionClosed
+            if (Interlocked.Exchange(ref _connectionOpenedStatus, 2) == 1)
```

Comment:
Is this intended to be always on? If not would it make sense if telemetry is enabled before doing atomic inc?

Reply:
I will add an IsEnabled check here